### ISSUE TITLE
Adding a tag which includes trace hex ID

### ISSUE THREAD

**Issue by indrekj** [open-telemetry/opentelemetry-collector-contrib#6426]
**Is your feature request related to a problem? Please describe.**

We're using DataDog APM but we're using our own hosted Kibana. The recommended approach by DataDog was to add a logs link to the traces so it would be possible to quickly navigate to Kibana from the DD UI. Something like (`https://logs?q...trace_id:$HEX_TRACE_ID`). To my knowledge, this currently is not possible.

My initial idea was to enhance `attributesprocessor` to be able to insert tags with a prefix. I even tried to [implement it](https://github.com/indrekj/opentelemetry-collector-contrib/commit/d500d170753ee776b38736e789dcf473ce873710), but at some point, I noticed that trace ID probably isn't available at that moment.

Datadog exporter itself sends `otlp.trace_id` tag, but that's [added in the exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/ff848aff8eeb90a0112764059a7bbeb1d1a684a0).

**Describe the solution you'd like**

Maybe it would somehow be possible to add a tag with Trace ID in the `attributesprocessor`. Another not that generic solution would be to have this as part of the datadog exporter.

**Describe alternatives you've considered**

This probably could also be done on the application layer, but then every application has to be changed which is not really an option.

**Comment by jpkrohling** [comment]
> Maybe it would somehow be possible to add a tag with Trace ID in the attributesprocessor

This should be doable. The commit you linked is attempting to add a prefix, perhaps you can just add a trace ID? Have you tried that?

**Comment by indrekj** [comment]
The attributes don't include trace ID. E.g., one of the requests has these attributes `map[http.method:GET http.status_code:404 http.url:/engagement_requests]`.

> commit you linked is attempting to add a prefix, perhaps you can just add a trace ID?

Do you mean like a new configuration option: `AppendTraceId: true`?

**Comment by jpkrohling** [comment]
I mean, an option to include the trace ID as an attribute to the spans.

**Comment by indrekj** [comment]
Sorry, I'm still confused about the proposed solution. (also very new to golang)

[a.attrProc.Process()](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/c3f0a18341b728bed91ddd0f303b3f612722bd4f/processor/attributesprocessor/attributes_trace.go#L59) is also used by [resourceprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/441dbd2ec131d61e39e62cb9e583a350afb43f45/processor/resourceprocessor/resource_processor.go#L40) and by [attributes_log](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/441dbd2ec131d61e39e62cb9e583a350afb43f45/processor/attributesprocessor/attributes_log.go#L59).

I could pass in trace_id like:
```
				extra := make(map[string]string)
				extra["trace_id"] = span.TraceID().HexString()
				a.attrProc.Process(span.Attributes(), extra)
```
and in the other two places I could use an empty map. This way it would be agnostic to the span struct.

With this, I could create a configuration interface like:

```
    - key: logs_url
      action: insert
      value: "https://logs?trace_id="
      append: trace_id # allows keys that were passed in through the extra map
```

Basically, I think there are two main problems:
1. Getting Trace ID to the attraction module. Would it be fine to pass these as a second argument like shown above?
2. Appending or Prepending the Trace ID to an actual URL


Theoretically, I could add trace_id directly to `span.Attributes()`, but then I'd need to modify that key or remove it after `logs_url` key is created. My end goal is to get a `logs_url` attribute which itself includes trace id as a query parameter. I hope this makes sense.

**Comment by jpkrohling** [comment]
> Sorry, I'm still confused about the proposed solution. (also very new to golang)

I thought you didn't have access to the trace ID in the backend you are using, but re-reading the original description, you do seem to have it.

About adding a `logs_url` to the spans: not sure this would be a good idea. This is certainly going to cause your spans to become larger, having a considerable effect on your systems/backend, given that you are adding it to every span.

> Would it be fine to pass these as a second argument like shown above?

Good point, they are all receiving the generic attributes, not the specific data point. I'm not sure I like the idea of adding the trace ID as the second attribute, though.

@bogdandrutu, @codeboten, do you have any ideas?

**Comment by indrekj** [comment]
> not sure this would be a good idea

I do agree, but that was the suggestion by the DataDog team. There's currently no other way to move from the traces view to kibana otherwise in the DD. Maybe it would be possible to add that link only to the root span, but that would make this change even more complicated.

**Comment by indrekj** [comment]
@KSerrania @mx-psi @gbbr @knusbaum (DataDog exporter codeowners), would you accept a PR to the datadog exporter to add logs URL only to the **root** spans? Basically an exporter setting `logs_url` which then is appended by the trace id?

**Comment by mx-psi** [comment]
Looks like we missed this. Sorry for the very late response, but is this still a problem @indrekj ?

**Comment by indrekj** [comment]
We ended up with not adding the logs link at all. Basically we're just copying the trace ID and pasting it to Kibana. Not the most efficient solution but we're kind of used to it now already. So unless somebody else needs it I'm okay if this is closed.

**Comment by mx-psi** [comment]
Okay, we can keep this open in case someone else shows up :)
